pkg/services/baremetal/baremetal/baremetal.go

```diff
@@ -673,17 +673,27 @@ func (s *Service) setProviderID(ctx context.Context) error {
 		return fmt.Errorf("host not found for machine %s: %w", s.scope.Machine.Name, err)
 	}
 
-	if host.Spec.Status.ProvisioningState != infrav1.StateProvisioned {
+	// set providerID in phase ensure provisioned and provisioned as from this point on the node might join the cluster
```

Comment:
@janiskemper I suggest to use quotes to make it more readable: 

> set providerID in phase "ensure provisioned" and "provisioned" ...

Reply:
fixed